src/formatter-settings/index.ts

```diff
@@ -4,27 +4,347 @@
+                    const value = settings[j].getAttribute("value");
+                    if (!id || !value) {
+                        continue;
+                    }
+                    this.profileElements.set(id, settings[j]);
+                    this.profileSettings.set(id, value);
+                    this.profileSettingsArray.push(settings[j]);
+                }
+                break;
+            }
+        }
+        if (this.profileElements.size === 0) {
+            throw new Error("No valid settings exist in the profile");
+        }
+        for (const setting of initializeSupportedProfileSettings(Number(this.settingsVersion))) {
+            if (this.profileSettings.has(setting.id)) {
+                let value = this.profileSettings.get(setting.id);
+                if (!value) {
+                    value = setting.value;
+                }
+                const webViewValue: string | undefined = FormatterConverter.profile2WebViewConvert(setting.id, value);
+                if (!webViewValue) {
+                    throw new Error(`Invalid setting value in ${setting.id}`);
+                }
+                setting.value = webViewValue;
+            }
+            webviewPanel.webview.postMessage({
+                command: "loadSetting",
+                setting: setting,
+            });
+        }
+    }
+
+    private async inheritVSCodeSettings(webviewPanel: vscode.WebviewPanel) {
+        const javaTabSize: number | undefined = await this.findJavaSpecificSetting<number>("editor.tabSize");
+        const javaSpacePolicy: boolean | undefined = await this.findJavaSpecificSetting<boolean>("editor.insertSpaces");
+        const vscodeTabSize: number | undefined = vscode.workspace.getConfiguration().get<number>("editor.tabSize");
+        const vscodeSpacePolicy: boolean | undefined = vscode.workspace.getConfiguration().get<boolean>("editor.insertSpaces");
+        const tabSize = (javaTabSize === undefined) ? vscodeTabSize : javaTabSize;
+        const spacePolicy = (javaSpacePolicy === undefined) ? vscodeSpacePolicy : javaSpacePolicy;
+        for (const setting of initializeSupportedVSCodeSettings()) {
+            if (setting.id === SupportedSettings.TABULATION_CHAR) {
+                setting.value = (spacePolicy === false) ? "tab" : "space";
+                this.profileSettings.set(setting.id, setting.value);
+            } else if (setting.id === SupportedSettings.TABULATION_SIZE) {
+                setting.value = String(tabSize);
+                this.profileSettings.set(setting.id, setting.value);
+            }
+            webviewPanel.webview.postMessage({
+                command: "loadSetting",
+                setting: setting,
+            });
+        }
+    }
+
+    private async findJavaSpecificSetting<T>(setting: string): Promise<T | undefined> {
+        let result: T | undefined;
+        const config = vscode.workspace.getConfiguration().get<Object>("[java]");
```

Comment:
Why not use `{ languageId: "java" }` as scope

Reply:
Here we can use
```
const config = vscode.workspace.getConfiguration("editor", { languageId: "java" });
const tabSize = config.get("tabSize")
```
to get the value.